app/map/controllers.js

```diff
@@ -0,0 +1,36 @@
+const axios = require('axios')
+const qs = require('qs')
+
+const {
+  MAPPING: { AUTH_URL: authUrl, API_SECRET: apiSecret, API_KEY: apiKey },
+} = require('../../config')
+
+const token = function (req, res) {
+  return axios
+    .request({
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/x-www-form-urlencoded',
+      },
+      url: authUrl,
+      auth: {
+        username: apiKey,
+        password: apiSecret,
+      },
+      data: qs.stringify({ grant_type: 'client_credentials' }),
+    })
+    .then(response => {
+      res.type('json')
```

Comment:
Do we need `.type('json')` as well as `.json()`?

These can also be chained:

```javascript
res
  .status(200)
  .json(response.data)
```

Reply:
Ah, I didn't notice this in the documentation:
> This method sends a response (**with the correct content-type**) 

Reason why they weren't chained was to make the tests easier, otherwise I'd have to return a thing that returned a spy, and that was much more fragile than having two separate spies that were checked.